File: api/src/main/java/mock/backend/api/ws/user/UserWebService.java
package mock.backend.api.ws.user;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseStatus;

/**
 * Created by semihunaldi on 23.08.2016.
 */

@RequestMapping(path = "/api")
public interface UserWebService
{
    @RequestMapping(value = "/user/queryUserByEmail", method = {RequestMethod.GET})
    //http://localhost:8080/api/user/queryUserByEmail?email=test@test.com
    public UserDTO queryUserByEmail(@RequestParam(value = "email", defaultValue = "") String email);

    @RequestMapping(value = "/user/queryUserById", method = {RequestMethod.GET})
    //http://localhost:8080/api/user/queryUserById?id=
    public UserDTO queryUserById(@RequestParam(value = "id", defaultValue = "") String id);

    @RequestMapping(value = "/user/queryUserByUserName", method = {RequestMethod.GET})
    //http://localhost:8080/api/user/queryUserByUserName?userName=
    public UserDTO queryUserByUserName(@RequestParam(value = "userName", defaultValue = "") String userName);

    @RequestMapping(value = "/user/createUser", method = {RequestMethod.POST})
    @ResponseStatus(value = HttpStatus.OK )
    //http://localhost:8080/api/user/createUser
    public CreateUserDTO createUser(@RequestBody CreateUserDTO createUserDTO);

    @RequestMapping(value = "/user/updateUser", method = {RequestMethod.POST})
    @ResponseStatus(value = HttpStatus.OK )
    //http://localhost:8080/api/user/updateUser
    public CreateUserDTO updateUser(@RequestBody CreateUserDTO createUserDTO);
}

File: api/src/main/kotlin/mock/backend/api/ws/user/UserWebService.kt
package mock.backend.api.ws.user

import mock.backend.api.ws.user.model.CreateUserDTO
import mock.backend.api.ws.user.model.UserDTO
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RequestMapping(path = arrayOf("/api"))
interface UserWebService
{
    @RequestMapping(value = "/user/queryUserByEmail", method = arrayOf(RequestMethod.GET))
    abstract //http://localhost:8080/api/user/queryUserByEmail?email=test@test.com
    fun queryUserByEmail(@RequestParam(value = "email", defaultValue = "") email: String): UserDTO

    @RequestMapping(value = "/user/queryUserById", method = arrayOf(RequestMethod.GET))
    abstract //http://localhost:8080/api/user/queryUserById?id=
    fun queryUserById(@RequestParam(value = "id", defaultValue = "") id: String): UserDTO

    @RequestMapping(value = "/user/queryUserByUserName", method = arrayOf(RequestMethod.GET))
    abstract //http://localhost:8080/api/user/queryUserByUserName?userName=
    fun queryUserByUserName(@RequestParam(value = "userName", defaultValue = "") userName: String): UserDTO

    @RequestMapping(value = "/user/createUser", method = arrayOf(RequestMethod.POST))
    @ResponseStatus(value = HttpStatus.OK)
    abstract //http://localhost:8080/api/user/createUser
    fun createUser(@RequestBody createUserDTO: CreateUserDTO): CreateUserDTO

    @RequestMapping(value = "/user/updateUser", method = arrayOf(RequestMethod.POST))
    @ResponseStatus(value = HttpStatus.OK)
    abstract //http://localhost:8080/api/user/updateUser
    fun updateUser(@RequestBody createUserDTO: CreateUserDTO): CreateUserDTO
}